app/scripts/metamask-controller.js

```diff
@@ -3856,6 +3856,16 @@ export default class MetamaskController extends EventEmitter {
           this.approvalController.addAndShowApprovalRequest.bind(
             this.approvalController,
           ),
+        startApprovalFlow: this.approvalController.startFlow.bind(
```

Comment:
Just for audit sake, my understanding is that this is the correct place for our internal hooks used by the middleware but not to be called from the dApps.

As I understand, anything that can be called by the dApps would instead be defined in `this.metamaskMiddleware`.

Reply:
Yes. I just added them here because that's the only place where they'll be called from for the foreseeable future.

Should we already add the ability to call them from dapps even before having a use case?